Documentation/data-cleanup/python-script/dashboard_resource_helper.py

```
@@ -0,0 +1,152 @@
+        if (not force_refresh and 
+            self.collection_cache is not None and 
+            now - self.last_cache_time < self.cache_ttl):
+            return self.collection_cache
+
+        url = f"{self.metabase_url}/collection"
+        headers = {
+            "Content-Type": "application/json",
+            "X-Metabase-Session": self.get_session_token(),
+        }
+
+        resp = requests.get(url, headers=headers)
```

Ruff (0.15.5)

[error] Probable use of `requests` call without timeout

(S113)